gapis/replay/wait_for_fence.go

```diff
@@ -17,27 +17,56 @@ package replay
 import (
 	"context"
 
+	"github.com/google/gapid/gapir"
 	"github.com/google/gapid/gapis/api"
 	"github.com/google/gapid/gapis/api/transform"
 	"github.com/google/gapid/gapis/replay/builder"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/trace"
 )
 
-type WaitForFence struct{}
+type WaitForFence struct {
+	TraceOptions  *service.TraceOptions
+	SignalHandler *SignalHandler
+}
 
 func (t *WaitForFence) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
 	if id == 0 {
-		t.AddWaitInstruction(ctx, id, out)
+		t.AddStartPerfetto(ctx, id, out)
 	}
 	out.MutateAndWrite(ctx, id, cmd)
 }
 
-func (t *WaitForFence) Flush(ctx context.Context, out transform.Writer)    {}
+func (t *WaitForFence) Flush(ctx context.Context, out transform.Writer) {
+	t.AddStopPerfetto(ctx, out)
+}
+
 func (t *WaitForFence) PreLoop(ctx context.Context, out transform.Writer)  {}
 func (t *WaitForFence) PostLoop(ctx context.Context, out transform.Writer) {}
 
-func (t *WaitForFence) AddWaitInstruction(ctx context.Context, id api.CmdID, out transform.Writer) {
+func (t *WaitForFence) AddStartPerfetto(ctx context.Context, id api.CmdID, out transform.Writer) {
```

Comment:
As discussed offline -- this is tangling perfettoisms with the mechanism to insert waits and get callbacks from them.

Reply:
Pulled the perfetto jazz into a wrapper function